Discovery/ConfiguredClientsStrategy.php

```diff
@@ -24,24 +25,37 @@ class ConfiguredClientsStrategy implements DiscoveryStrategy, EventSubscriberInt
     private static $client;
 
     /**
-     * @param HttpClient $httpClient
+     * @var HttpAsyncClient
      */
-    public function __construct(HttpClient $httpClient)
+    private static $asyncClient;
+
+    /**
+     * @param HttpClient      $httpClient
+     * @param HttpAsyncClient $asyncClient
+     */
+    public function __construct(HttpClient $httpClient = null, HttpAsyncClient $asyncClient = null)
     {
         static::$client = $httpClient;
+        static::$asyncClient = $asyncClient;
     }
 
     /**
      * {@inheritdoc}
      */
     public static function getCandidates($type)
     {
-        if (static::$client !== null && $type == HttpClient::class) {
+        if (static::$client !== null && $type === HttpClient::class) {
```

Comment:
Should the type check rather come first? This way the first condition is evaluated even if the type is a MessageFactory.